test/feature_discovery_test.dart

```diff
@@ -92,28 +92,41 @@ void main() {
       'featureIdC',
     ];
     final List<String> texts = textsToMatch(steps);
-    testWidgets('Two overlays have the same featureId',
-        (WidgetTester tester) async {
-      await tester.pumpWidget(const TestWidget(featureIds: featureIds));
-      final Finder finder = find.byType(TestIcon);
-      expect(finder, findsNWidgets(featureIds.length));
-      final BuildContext context = tester.firstState(finder).context;
-      texts.forEach((t) => expect(find.text(t), findsNothing));
-      FeatureDiscovery.discoverFeatures(context, steps);
-      await tester.pumpAndSettle();
-      // First overlay should appear.
-      expect(find.text(texts[0]), findsOneWidget);
-      FeatureDiscovery.completeCurrentStep(context);
-      await tester.pumpAndSettle();
-      // First overlay should have disappeared, and overlays 2 and 3 should be displayed.
-      expect(find.text(texts[0]), findsNothing);
-      expect(find.text(texts[1]), findsNWidgets(2));
-      FeatureDiscovery.completeCurrentStep(context);
-      await tester.pumpAndSettle();
-      // Overlays 2 and 3 should have disappeared, and the last overlay should appear.
-      expect(find.text(texts[1]), findsNothing);
-      expect(find.text(texts[2]), findsOneWidget);
-    });
+
+    for (final bool allowShowingDuplicate in <bool>[true, false]) {
```

Comment:
I suppose that this is what the test was intended to look like the whole time, right?

Anyway, it fits really nice to be able to tests these two scenarios 👍